### PR TITLE
Assume Riff-Raff role within the action itself

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

This PR removes the requirement to use the [configure-aws-credentials]( https://github.com/aws-actions/configure-aws-credentials) Github Action in order to assume the Riff-Raff role. We're now using the AWS SDK to assume the role directly within this action.

This introduces a breaking change, by requiring a new field `roleArn`, so that callers can specify the ARN of the Riff-Raff role.

## How to test

This was tested by creating a [draft PR](https://github.com/guardian/service-catalogue/pull/787) on a repo that uses this action (leveraging the fact we can supply actions by their commit SHA), removing the `configure-aws-credentials` step and passing in the role ARN. We confirmed that artifacts were successfully uploaded to Riff-Raff.

## How can we measure success?

We no longer have AWS credentials stored as environment variables.

## Have we considered potential risks?

This should be fairly low-risk, as upgrading should require CI passing in a branch before merging to main.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -10,6 +11,11 @@ import type { Deployment } from './riffraff';
 import { manifest, riffraffPrefix } from './riffraff';
 import { S3Store, sync } from './s3';
 
+/**
+ * Amazon STS expects OIDC tokens with the `aud` (audience) field set to `sts.amazonaws.com`
+ */
+const GITHUB_OIDC_AUDIENCE = 'sts.amazonaws.com';
```
Comment (chrislomaxjones) [inline src/index.ts:17]: [This value is the default used as as input by the `configure-aws-credentials` action](https://github.com/aws-actions/configure-aws-credentials/blob/main/action.yml#L33-L36).

**Review by akash1810 — COMMENTED** [review]
Nice!

It looks like the pattern for retrieving inputs is to [get them early](https://github.com/guardian/actions-riff-raff/blob/8d7305be68e8c592ed4ec4bc3b80351e889ffb26/src/index.ts#L47)/fail fast. Worth doing the same for the `roleArn`?

It looks like the AWS action adds some [useful tags](https://github.com/aws-actions/configure-aws-credentials/blob/2e34541e3b5995d25a0cede76ca096af7e454763/src/assumeRole.ts#L102-L110) when assuming a role, in particular the `Repository` tag. Would we want to do something similar? I imagine it'll be useful for audit trails.

**Comment by AshCorr** [comment]
> It looks like the AWS action adds some [useful tags](https://github.com/aws-actions/configure-aws-credentials/blob/2e34541e3b5995d25a0cede76ca096af7e454763/src/assumeRole.ts#L102-L110) when assuming a role, in particular the `Repository` tag. Would we want to do something similar? I imagine it'll be useful for audit trails.

Hmm... This is a bit strange, I've tried to add tags but the STS SDK that we're using doesn't seem to support them... Upon further investigation it also appears that the old `aws-configure-credentials` action seems to not support them either when using the OIDC flow https://github.com/aws-actions/configure-aws-credentials/blob/2e34541e3b5995d25a0cede76ca096af7e454763/src/assumeRole.ts#L11

**Comment by AshCorr** [comment]
Aha!

As per https://docs.aws.amazon.com/IAM/latest/UserGuide/id_session-tags.html#id_session-tags_adding-assume-role-idp 

> To pass session tags from OpenID Connect (OIDC), you must include the session tags in the JSON Web Token (JWT).

It looks like the tags must be set inside the OIDC token, so only Github would be able to configure them.

**Comment by akash1810** [comment]
The PR description suggests these changes are aimed at improving the DX of using this action. IIUC this change was made during [MavenGate](https://blog.oversecured.com/Introducing-MavenGate-a-supply-chain-attack-method-for-Java-and-Android-applications/) remediation? Is it worth describing how this change works to mitigate MavenGate?

**Pull request merged**